packages/state-transition/src/block/processDeposit.ts

```diff
@@ -41,68 +49,132 @@ export function processDeposit(fork: ForkSeq, state: CachedBeaconStateAllForks,
+  fork: ForkSeq,
+  state: CachedBeaconStateAllForks,
+  pubkey: BLSPubkey,
+  withdrawalCredentials: Bytes32,
+  amount: UintNum64
+): void {
+  const {validators, epochCtx} = state;
+  // add validator and balance entries
+  const effectiveBalance =
+    fork < ForkSeq.electra ? Math.min(amount - (amount % EFFECTIVE_BALANCE_INCREMENT), MAX_EFFECTIVE_BALANCE) : 0;
+  validators.push(
+    ssz.phase0.Validator.toViewDU({
+      pubkey,
+      withdrawalCredentials,
+      activationEligibilityEpoch: FAR_FUTURE_EPOCH,
+      activationEpoch: FAR_FUTURE_EPOCH,
+      exitEpoch: FAR_FUTURE_EPOCH,
+      withdrawableEpoch: FAR_FUTURE_EPOCH,
+      effectiveBalance,
+      slashed: false,
+    })
+  );
 
-    const validatorIndex = validators.length - 1;
-    // Updating here is better than updating at once on epoch transition
-    // - Simplify genesis fn applyDeposits(): effectiveBalanceIncrements is populated immediately
-    // - Keep related code together to reduce risk of breaking this cache
-    // - Should have equal performance since it sets a value in a flat array
-    epochCtx.effectiveBalanceIncrementsSet(validatorIndex, effectiveBalance);
+  const validatorIndex = validators.length - 1;
+  // TODO Electra: Review this
+  // Updating here is better than updating at once on epoch transition
+  // - Simplify genesis fn applyDeposits(): effectiveBalanceIncrements is populated immediately
+  // - Keep related code together to reduce risk of breaking this cache
+  // - Should have equal performance since it sets a value in a flat array
+  epochCtx.effectiveBalanceIncrementsSet(validatorIndex, effectiveBalance);
 
-    // now that there is a new validator, update the epoch context with the new pubkey
-    epochCtx.addPubkey(validatorIndex, pubkey);
+  // now that there is a new validator, update the epoch context with the new pubkey
+  epochCtx.addPubkey(validatorIndex, pubkey);
 
-    // Only after altair:
-    if (fork >= ForkSeq.altair) {
-      const stateAltair = state as CachedBeaconStateAltair;
+  // Only after altair:
+  if (fork >= ForkSeq.altair) {
+    const stateAltair = state as CachedBeaconStateAltair;
 
-      stateAltair.inactivityScores.push(0);
+    stateAltair.inactivityScores.push(0);
 
-      // add participation caches
-      stateAltair.previousEpochParticipation.push(0);
-      stateAltair.currentEpochParticipation.push(0);
-    }
-  } else {
-    // increase balance by deposit amount
-    increaseBalance(state, cachedIndex, amount);
+    // add participation caches
+    stateAltair.previousEpochParticipation.push(0);
+    stateAltair.currentEpochParticipation.push(0);
+  }
+
+  if (fork < ForkSeq.electra) {
+    state.balances.push(amount);
+  } else if (fork >= ForkSeq.electra) {
+    state.balances.push(0);
+    const stateElectra = state as CachedBeaconStateElectra;
+    const pendingBalanceDeposit = ssz.electra.PendingBalanceDeposit.toViewDU({
+      index: validatorIndex,
+      amount: BigInt(amount),
```

Comment:
is it needed to model `amount` as BigInt? in `processPendingBalanceDeposits()` we need to cast it to Number anyway:
```typescript
for (const deposit of state.pendingBalanceDeposits.getAllReadonly()) {
    const {amount} = deposit;
    if (processedAmount + amount > availableForProcessing) {
      break;
    }
    increaseBalance(state, deposit.index, Number(amount));
    processedAmount = processedAmount + amount;
    nextDepositIndex++;
  }
```